### PR TITLE
`Feature`: Test View in Grading Edition

### PR DESCRIPTION
**Pull request by dfuchss**
Adding a test overview for the grading edition.

### For Review
As part of my work I've refactored many classes since I saw wrong package names.
Therefore, you should focus on these 5 classes (as they contain the actual work):

*    bundles/edu.kit.kastel.sdq.eclipse.common.api/src/edu/kit/kastel/sdq/eclipse/common/api/artemis/mapping/ResultsDTO.java
*    bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/AbstractResultTab.java
*    bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/ResultTabUI.java
*    bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/assessment/ArtemisGradingView.java
*    bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/assessment/ResultTab.java

I've triggered a build of the eclipse products here: https://github.com/kit-sdq/programming-lecture-eclipse-artemis/actions/runs/2163667344

## Current State

image

### CODE REVIEW

**Comment by dfuchss** [comment]
Note: This PR contains the basic functionality. The refactoring in #186 contains the enhancements.

**Comment by Shirkanesi** [comment]
@dfuchss I've got problems when testing this feature.
It seems to work flawlessly when using the backlog, however `Start Correction Round 1` does cause an IllegalArgumentException because `result.resultString` is null (in AbstractResultTab line 87).

When this exception is thrown, the state-machine seems to end in an unintended state, where "reload", "save", "submit" won't work but "start correction round 1" and "refresh artemis state" will be available. You can only recover from this state by restarting eclipse and loading the submission from the backlog.

**Comment by dfuchss** [comment]
Yes, I'm currently fixing that in #186 
I've also encountered some issues

**Review by dfuchss — COMMENTED** [review]
Important classes for review (other classes are simply renamings or code improvements (sorry for the large PR)

* bundles/edu.kit.kastel.sdq.eclipse.common.api/src/edu/kit/kastel/sdq/eclipse/common/api/artemis/mapping/ResultsDTO.java
* bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/AbstractResultTab.java
* bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/ResultTabUI.java
* bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/assessment/ArtemisGradingView.java
* bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/assessment/ResultTab.java

File: bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/AbstractResultTab.java [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+		String success = this.calculateSuccessMessage(feedback);
+		int colorIndicator = this.calculateSuccessColorIndicator(feedback);
+
+		final TableItem item = new TableItem(table, SWT.NULL);
+		item.setData(feedback);
+		item.setText(0, name);
+		item.setText(1, "" + roundedCredits);
+		item.setText(2, success);
+		item.setForeground(2, Display.getDefault().getSystemColor(colorIndicator));
+		item.setText(3, feedback.getDetailText() != null ? CHECK_MARK_IN_UTF8 : X_MARK_IN_UTF8);
```
Comment (Shirkanesi) [inline bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/AbstractResultTab.java:146]: I'm not a huge fan of the checkmark / cross indicator. Seems quite unintuitive to have a checkmark even if your test failed.

File: bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/TestDetailsDialog.java [hunk 2]
```diff
@@ -13,7 +13,7 @@
 import org.eclipse.swt.widgets.Shell;
 import org.eclipse.swt.widgets.Text;
 
-public class TestDetailsDialog extends Dialog {
+class TestDetailsDialog extends Dialog {
 	private static final int WIDTH = 300;
```
Comment (Shirkanesi) [inline bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/TestDetailsDialog.java:17]: Does this size actually look good on 4k-screens?

File: bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/AbstractResultTab.java (hunk 1 above)
Comment (dfuchss) [reply bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/AbstractResultTab.java:146]: Any suggestions? I'm very happy with other indicators :)

File: bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/TestDetailsDialog.java (hunk 2 above)
Comment (dfuchss) [reply bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/TestDetailsDialog.java:17]: It's looks ok as far as I can see. The dialog itself is resizable (or it should). But if we want other sizes as default, we can do this.

File: bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/AbstractResultTab.java (hunk 1 above)
Comment (Shirkanesi) [reply bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/AbstractResultTab.java:146]: Maybe just "present" or something like that. I'll think about it, it's not a big problem but feels just a bit counterintuitive.

**Review by majuwa — APPROVED** [review]
Works for my test student with Artemis

**Pull request merged**